### PR TITLE
Jetty 9.4 follow-up

### PR DESCRIPTION
**Pull request by xvrl**
* restore proper error dispatch test removed in #71 
* switch from deprecated GzipFilter to GzipHandler
* add integration test for gzip encoding

### CODE REVIEW

**Comment by ijuma** [comment]
Are these fixes or improvements? As in, was any behaviour actually broken or it's just a code cleanliness issue (which is good, of course).

**Comment by xvrl** [comment]
@ijuma removing deprecated classes

File: core/src/test/java/io/confluent/rest/GzipHandlerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 Confluent Inc.
```
Comment (ewencp) [inline core/src/test/java/io/confluent/rest/GzipHandlerIntegrationTest.java]: lol, doesn't really matter but i think your header template is out of date

Comment (xvrl) [reply core/src/test/java/io/confluent/rest/GzipHandlerIntegrationTest.java]: Ha, the Gzip filter was added in 2016 so we can say it’s keeping with the spirit of the code :)

**Pull request merged**